keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,860 @@
+  - verify whether a job has a failed condition if the job meets to both successPolicy and terminating policies
+  in the same reconcile cycle
+
+- `k8s.io/kubernetes/pkg/controller/job`: `2 February 2024` - `91.5%`
+- `k8s.io/kubernetes/pkg/apis/batch/validation`: `2 February 2024` - `98.0%`
+
+##### Integration tests
+
+- Test scenarios:
+  - enabling, disabling and re-enabling of the `JobSuccessPolicy` feature gate
+  - handling of successPolicy when all indexes succeeded
+  - handling of the `.spec.successPolicy.criteria.succeededIndexes` when some indexes remain pending
+  - handling of the `.spec.successPolicy.criteria.succeededCount` when some indexes remain pending 
+  - handling of successPolicy when some indexes of job with `backOffLimitPerIndex` fail
+
+##### e2e tests
+
+- Test scenarios:
+  - handling of successPolicy when all indexes succeeded
+  - handling of the `.spec.successPolicy.criteria.succeededIndexes` when some indexes remain pending
+  - handling of the `.spec.successPolicy.criteria.succeededCount` when some indexes remain pending
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind the `JobSuccessPolicy` feature gate.
+- Unit and integration tests passed as designed in [TestPlan](#test-plan).
+
+#### Beta
+
+- E2E tests passed as designed in [TestPlan](#test-plan).
+- Added a new reasons to the existing `job_finished_total` metric in [Monitoring Requirements](#monitoring-requirements).
+- Feature is enabled by default.
+- Address all issues reported by users.
+- Decided whether we introduce a `whenCriterionAchived=continue` and `whenCriterionAchived=continueWithRecreations`.
+Please see the [Possibility for the lingering pods to continue running after the job meets the successPolicy](#possibility-for-the-lingering-pods-to-continue-running-after-the-job-meets-the-successpolicy)
+section for discussion points.
+- Decided whether we introduce a new CronJob concurrentPolicy, `ForbidUntilJobSuccessful`.
+Please see the [Possibility for introducing a new CronJob concurrentPolicy, "ForbidUntilJobSuccessful"](#possibility-for-introducing-a-new-cronjob-concurrentpolicy-forbiduntiljobsuccessful)
+section for more details.
+
+#### GA
+
+- No negative feedback.
+
+### Upgrade / Downgrade Strategy
+
+- Upgrade
+  - If the feature gate is enabled, `JobSuccessPolicy` are allowed to use only.
+  - If the feature gate is enabled without `JobSuccessPolicy`, 
+  the default values will be applied to a job object.
+  - Even if the feature gate is enabled, the Job controller doesn't update
+  `.status.conditions` in already finished jobs.
+- Downgrade
+  - Previously configured values will be ignored, and the job will be marked
+  as completed only when all indexes succeed.
+  - the Job controller doesn't update `.status.conditions` in already finished jobs.
+
+### Version Skew Strategy
+
+The apiserver's version should be consistent with the kube-controller-manager version,
+or this feature will not work.
```

Comment:
You can't guarantee that.

Reply:
It makes sense. I guess that we can put the same description as "BackoffLimitPerIndex".

Reply:
Done.